include/stl2/view/drop_while.hpp

```diff
@@ -54,7 +53,21 @@ STL2_OPEN_NAMESPACE {
 				using cache_t = typename drop_while_view::non_propagating_cache;
 				auto& iterator_self = static_cast<cache_t&>(*this);
 				if (!iterator_self) {
-					iterator_self = __stl2::find_if_not(base_, pred());
+					iterator_self = __stl2::find_if_not(base_, [this](auto&& i) mutable {
+#ifndef NDEBUG
+						// A predicate must be equality-preserving. While it's not possible to generally
+						// check that te predicate isn't equality-preserving, we can trap
+						// non-equality-preserving invocables on-the-spot by calling them multiple times
+						// and compare the results.
+						// This is only enabled in debug-mode, since it's potentially an expensive check.
+						auto result1 = get()(i);
+						auto result2 = get()(i);
+						STL2_ASSERT(result1 == result2);
+						return result1;
+#else
+						return get()(i);
```

Comment:
I think we can avoid the `#if` and simplify this all down to:
```c++
bool result = get()(i);
STL2_ASSERT(result == get()(i));
return result;
```
`Boolean` requires the implicit conversion to `bool` (and comparison with `bool`) to work, and we don't need to worry that holding a `bool` for a bit and then returning it is less efficient than might be the case for an arbitrary model of `Boolean` returned by a predicate.